### DISCUSSION TITLE
Update metadata after creating table.

### DISCUSSION THREAD

**Discussion by AliyevH** [sqlalchemy/alembic#1131] [Ideas]
Hi. After creating table, i see that metadata object doesn't updates itself.
What do you say about adding such feature?
Now i manually reflect metadata object to get created tables.

**Comment by CaselIT** [comment]
Hi,

I'm not sure what you mean by 
>  see that metadata object doesn't updates itself.

How should the metadata be updated?

**Reply by AliyevH** [reply]
For example. 
```
engine = create_engine(uri, echo=False)
session = Session(self.engine, autocommit=False, autoflush=False)
metadata = MetaData(bind=self.engine)
base = declarative_base()
base.metadata.reflect(self.engine)
```
First i reflect database. It means, now in metadata object i can view all tables that currently exists.
After creating new table. i can't get new table information from base.metadata

**Reply by zzzeek** [reply]
you can call reflect() again and it will add the new table(s).

metadata itself is only a set of table structures and is not linked to a specific database, with the exception of the ``bind=self.engine`` you have there; that's [deprecated](https://docs.sqlalchemy.org/en/14/changelog/migration_20.html#implicit-and-connectionless-execution-bound-metadata-removed) after many years of being [strongly discouraged](https://docs.sqlalchemy.org/en/14/core/connections.html#dbengine-implicit) and is not available in the next major release of SQLAlchemy.
